tests/test_aci_worker.py

```diff
@@ -0,0 +1,942 @@
+        "subscription_id": SecretStr("sub_id"),
+        "name": None,
+        "task_watch_poll_interval": 0.05,
+        "stream_output": False,
+    }
+
+    for k, v in overrides.items():
+        values = {**values, k: v}
+
+    container_instance_variables = AzureContainerVariables(**values)
+
+    json_config = {
+        "job_configuration": AzureContainerJobConfiguration.json_template(),
+        "variables": container_instance_variables.dict(),
+    }
+
+    container_instance_configuration = (
+        await AzureContainerJobConfiguration.from_template_and_values(
+            json_config, values
+        )
+    )
+
+    if run_prep:
+        container_instance_configuration.prepare_for_flow_run(worker_flow_run)
+
+    return container_instance_configuration
+
+
+def get_command_from_deployment_parameters(parameters):
+    deployment_arm_template = parameters.template
+    # We're only interested in the first resource, because our ACI
+    # flow run container groups only have a single container by default.
+    deployment_resources = deployment_arm_template["resources"][0]
+    deployment_properties = deployment_resources["properties"]
+    deployment_containers = deployment_properties["containers"]
+
+    command = deployment_containers[0]["properties"]["command"]
+    return command
+
+
+@pytest.fixture()
+def running_worker_container_group():
+    """
+    A fixture that returns a mock container group simulating a
+    a container group that is currently running a flow run.
+    """
+    container_group = create_mock_container_group(state="Running", exit_code=None)
+    container_group.provisioning_state = ContainerGroupProvisioningState.SUCCEEDED
+    return container_group
+
+
+@pytest.fixture()
+def completed_worker_container_group():
+    """
+    A fixture that returns a mock container group simulating a
+    a container group that successfully completed its flow run.
+    """
+    container_group = create_mock_container_group(state="Terminated", exit_code=0)
+    container_group.provisioning_state = ContainerGroupProvisioningState.SUCCEEDED
+
+    return container_group
+
+
+# Fixtures
+@pytest.fixture
+def aci_credentials(monkeypatch):
+    client_id = "test_client_id"
+    client_secret = "test_client_secret"
+    tenant_id = "test_tenant_id"
+
+    mock_credential = Mock(wraps=ClientSecretCredential, return_value=Mock())
+
+    monkeypatch.setattr(
+        prefect_azure.credentials,
+        "ClientSecretCredential",
+        mock_credential,
+    )
+
+    credentials = AzureContainerInstanceCredentials(
+        client_id=client_id, client_secret=client_secret, tenant_id=tenant_id
+    )
+
+    return credentials
+
+
+@pytest.fixture
+def aci_worker(mock_prefect_client, monkeypatch):
+    monkeypatch.setattr(
+        prefect_azure.workers.container_instance,
+        "get_client",
+        Mock(return_value=mock_prefect_client),
+    )
+    return AzureContainerWorker(work_pool_name="test_pool")
```

Comment:
Workers are intended to be run a context managers. There might be parts of the worker that aren't setup if they aren't used as a context manager.
```suggestion
    async with AzureContainerWorker(work_pool_name="test_pool") as worker:
        yield worker
```